cmake/helpers.cmake

```diff
@@ -36,6 +36,11 @@ function(add_umf_target_compile_options name)
         if(UMF_DEVELOPER_MODE)
             target_compile_options(${name} PRIVATE
                 -Werror
+                -Wextra
+                -Wshadow
```

Comment:
I don't think we should use `-Wshadow` (especially for C++) where it's perfectly fine to have class/struct members named the same way as e.g. function params.

Reply:
oneCCL enables this flag during compilation and if they did this and compiled UMF, they would get errors

Reply:
Ok, then perhaps we should add some comments why are we adding those flags?

Reply:
I wonder, shouldn't it be the third party repo's problem that it enforces some compiler flags globally on included repositories such as UMF? In the end, we can't control all repos and their flags.

That's the reason why we use modified versions of `add_executable` and `add_library`. By doing this, we don't enforce flags on eventual subprojects.